### ISSUE TITLE
Class for statevars and argsdef as dict

### ISSUE THREAD

**Issue by smaiLee** [flyte/upnpclient#4]
What do you think about a class for representing statevars?
Currently they are just a dict in another dict `Service.statevars`

And another related wish:
Statevars are held in `argsdef_in` and `argsdef_out` as a list of tuples of arg_name and the actual statevar.
Would you mind if I change argsdef_in and _out to dictionaries of `{ arg_name: statevar }`
This would make processing of an action result simpler (currently I have to convert the argsdef_out to a dict first).

**Comment by flyte** [comment]
Yeah, that could do with a bit of attention, you're right. I wonder if there's a way to include their uPnP 'type' so that we can provide (un)marshal and validation methods too.

I'm happy with changing `argsdef_(in|out)` to dictionaries. There's never a case where two arguments share the same key, so go for it.

**Comment by smaiLee** [comment]
I just did the work and refactored the argsdef lists to dicts and implemented the class `Statevar` with methods `validate_arg` and `marshal_value`.

It's completely untested yet as I currently have no UPnP devices around me (and not even a Python runtime, indeed). I will test it this evening and make a PR.
You can preview the argsdef dict at 5199a14c0d082f944ae8c705e83f4f0503ad6cb2 and the statevar class at 5f4c251c3f0d5d0c3e9b5e1367c50916ef8f10f8 if you like.

The `marshal_value` method just calls the existing marshal.py. Do you think it would be better to move the whole functionality to the method on `Statevar` and remove marshal.py?

Do you mind backward compatibility? Then I could add `__getitem__` to simulate dict-like value access like before: 
```
    def __getitem__(self, key):
        """
        Allow attributes to be returned as dictionary keys (for backward compatibility).
        """
        return getattr(self, key)
```

**Comment by flyte** [comment]
Good work! I had a quick look through just now - looks good. If you've got time, I'd appreciate it if you could update the Readme to reflect the changes, as it's now wrong. You may find there are some tests which fail now too.

I'd like to keep `marshal.py` I think, as it keeps that stuff in its own place. If anybody wants to use it for another reason then it makes it less cumbersome than having to use our classes.

I'm not that fussed about backward compatibility at the moment. Anyone can still use `getattr` if the param name is in a variable anyway, so it's probably not necessary.

PS. I'm on holiday for two weeks from Saturday, so if I stop replying, please don't get offended :)

**Comment by smaiLee** [comment]
I will have a look at Readme and examples and run the tests. As mentioned, this was completely "blind coding" as I could not even run it in Python.

I completely agree to use marshal.py.

Well, it's hard as I did already got used to your fast answering. :smile: But have a great holiday!
